pkg/monitors/monitor-service.go

```diff
@@ -12,10 +14,54 @@ type MonitorService interface {
 	GetByName(name string) (*models.Monitor, error)
 	Remove(m models.Monitor)
 	Setup(p config.Provider)
+	Equal(oldMonitor models.Monitor, newMonitor models.Monitor) bool
 }
 
 func CreateMonitorService(p *config.Provider) MonitorServiceProxy {
 	monitorService := (&MonitorServiceProxy{}).OfType(p.Name)
 	monitorService.Setup(*p)
 	return monitorService
 }
+
+func SetupMonitorServicesForProviders(providers []config.Provider) []MonitorServiceProxy {
+	if len(providers) < 1 {
+		log.Panic("Cannot Instantiate controller with no providers")
+	}
+
+	monitorServices := []MonitorServiceProxy{}
+
+	for index := 0; index < len(providers); index++ {
+		monitorServices = append(monitorServices, CreateMonitorService(&providers[index]))
+		log.Info("Configuration added for " + providers[index].Name)
+	}
+
+	return monitorServices
+}
+
+func SetupMonitorServicesForProvidersTest(providers []config.Provider) []MonitorServiceProxy {
+	if len(providers) < 1 {
+		log.Panic("Cannot Instantiate controller with no providers")
+	}
+	// TODO: Fix provider specific implementation and then add them to this list
+	allowedProviders := []string{"UptimeRobot", "StatusCake"}
```

Comment:
Are the rest providers not implemented with the operator pattern yet?

Reply:
The are implemented and integrated within the operator. They have just been disabled for this particular test case that i wrote against CRD reconciliation. For AppInsights, Pingdom and Uptime we don't have the appropriate credentials. And for Updown and GCloud there are issues with the existing implementation, that are there before the migration to operator. Had to restrict this test case to only a handful of providers.